pkg/solidity-utils/contracts/math/BasePoolMath.sol

```diff
@@ -156,11 +162,14 @@ library BasePoolMath {
         }
 
         // Calculate the new invariant with fees applied.
-        uint256 invariantWithFeesApplied = computeInvariant(newBalances);
+        // This invariant should be lower than the original one, so we don't need to check invariant ratio bounds again.
+        uint256 invariantWithFeesApplied = pool.computeInvariant(newBalances);
 
         // Calculate the amount of BPT to mint. This is done by multiplying the
         // total supply with the ratio of the change in invariant.
         // Since we multiply and divide we don't need to use FP math.
+        // Round down since we're calculating BPT amount out. `invariantWithFeesApplied` was rounded down before, which
```

Comment:
`invariantWithFeesApplied` was rounded down before

This statement here is slightly misleading. It's more that we round up on the taxable and swap fee amount, leading to lower balances being used to calculate the invariant.